src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,93 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    public static final int DEFAULT_SIZE = 16;
+    public static final double RESIZE_FACTOR = 0.75;
+    private Node<K, V>[] storage;
+    private int size;
+
+    public MyHashMap() {
+        storage = (Node<K, V>[]) new Node[DEFAULT_SIZE];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> existingNode = getNodeAt(key);
+        if (existingNode != null) {
+            existingNode.value = value;
+            return;
+        }
+        resize();
+        insertNode(new Node<>(key, value), storage);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        final Node<K, V> existingNode = getNodeAt(key);
+        if (existingNode == null) {
+            return null;
+        }
+        return existingNode.value;
+    }
+
+    private Node<K, V> getNodeAt(K key) {
```

Comment:
Let's put your private methods after public ones. Also, I'd recommend renaming this method (for example, 'getNodeForKey'), as its purpose isn't clear enough.